test/e2e/node_drain_test.go

```diff
@@ -34,6 +44,38 @@ var _ = Describe("When testing Node drain", func() {
 			SkipCleanup:            skipCleanup,
 			Flavor:                 ptr.To("topology"),
 			InfrastructureProvider: ptr.To("docker"),
+			VerifyNodeVolumeDetach: true,
+			AdditionalResources: func(clusterProxy framework.ClusterProxy, cluster *clusterv1.Cluster, _ *controlplanev1.KubeadmControlPlane, _ []*clusterv1.MachineDeployment) {
+				workloadClusterClient := clusterProxy.GetWorkloadCluster(ctx, cluster.Namespace, cluster.Name).GetClient()
+
+				nodeList := &corev1.NodeList{}
+				Expect(workloadClusterClient.List(ctx, nodeList)).To(Succeed())
+
+				for _, node := range nodeList.Items {
```

Comment:
nit: let's add a few lines describing what we are doing here (deploying a fake volume attachement for each node)